File: 8kyu/sum-without-highest-and-lowest-number/sum-without-highest-and-lowest-number.js
/* 
PREP:
- set the condition
- set sum function
- check the condition
   - if true, calc the sum of the array and then minus highest and lowest numbers
   - if false, return 0
*/

function sumArray(array) {
	const condition = Array.isArray(array) && array.length > 1;
	const sum = (acc, cur) => acc + cur;
	return condition
		? array.reduce(sum, 0) - Math.min(...array) - Math.max(...array)
		: 0;
}

File: 8kyu/sum-without-highest-and-lowest-number/sum-without-highest-and-lowest-number.py
def sum_array(arr):
  if arr == None or len(arr) < 2: return 0
  min_num, max_num = min(arr), max(arr)
  return sum(arr) - min_num - max_num

print(sum_array([]))
print(sum_array([6, 2, 1, 8, 10]))
print(sum_array([-6, -20, -1, -10, -12]))